### PR TITLE
e2e: check runtime permission changing

### PR DESCRIPTION
**Pull request by mitake**
This commit adds extends the test for checking runtime permission
grant/revoke.

/cc @heyitsanthony

### CODE REVIEW

File: e2e/ctl_v3_auth_test.go [hunk 1]
```diff
@@ -140,3 +179,64 @@ func authSetupTestUser(cx ctlCtx) {
 		cx.t.Fatal(err)
 	}
 }
+
+func ctlV3RoleGrantPermission(cx ctlCtx, rolename string, perm grantingPerm) error {
```
Comment (heyitsanthony) [inline e2e/ctl_v3_auth_test.go]: these role functions probably belong in ctl_v3_role_test.go

File: e2e/ctl_v3_auth_test.go [hunk 2]
```diff
@@ -111,6 +111,45 @@ func authCredWriteKeyTest(cx ctlCtx) {
 	if err := ctlV3Get(cx, []string{"foo"}, []kv{{"foo", "bar"}}...); err != nil {
 		cx.t.Fatal(err)
 	}
+
+	// try put to not granted key
+	cx.user, cx.pass = "test-user", "pass"
```
Comment (xiang90) [inline e2e/ctl_v3_auth_test.go]: this test is already HUGE. can we make another test for this?

File: e2e/ctl_v3_auth_test.go (hunk 1 above)
Comment (mitake) [inline e2e/ctl_v3_auth_test.go]: OK, I'll move the function to the role file.

File: e2e/ctl_v3_auth_test.go (hunk 2 above)
Comment (mitake) [inline e2e/ctl_v3_auth_test.go]: I agree. I'll create a new test for the scenario.

**Comment by mitake** [comment]
@xiang90 @heyitsanthony updated for your comments, PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by xiang90** [comment]
lgtm

**Pull request merged**